cpu/stm32f0/include/vendor/stm32f031x6.h

```diff
@@ -4967,6 +4967,9 @@ typedef struct
 #define USART_CR1_EOBIE_Pos           (27U)
 #define USART_CR1_EOBIE_Msk           (0x1U << USART_CR1_EOBIE_Pos)            /*!< 0x08000000 */
 #define USART_CR1_EOBIE               USART_CR1_EOBIE_Msk                      /*!< End of Block interrupt enable */
+#define USART_CR1_M1_Pos              (28U)
```

Comment:
I haven't checked in details using the reference manual (it takes some time!), but if it is not in the CMSIS already, there is a high chance that 7 bit is not supported for this platform.